src/OpenSearch.Client/Search/Search/Highlighting/Highlight.cs

```diff
@@ -107,8 +107,9 @@ public interface IHighlight
 		/// If this setting is set to a non-negative value, the highlighting stops at this defined maximum limit, and the
 		/// rest of the text is not processed, thus not highlighted and no error is returned.
 		/// </summary>
-		[DataMember(Name ="max_analyzed_offset")]
-		int? MaxAnalyzedOffset { get; set; }
+		/// <remarks>Introduced in OpenSearch 2.2</remarks>
+		[DataMember(Name ="max_analyzer_offset")]
```

Comment:
It should have both `max_analyzer_offset` and `max_analyzed_offset` for compatibility with pre 2.2 and post 2.2 versions.

Reply:
This option did not exist in OpenSearch pre 2.2 with either spelling, so the option would not have functioned at all before. It only would have working if using the client against ES versions that have it with the `ed` spelling it but we make no guarantees of compatibility with ES. Which is why I'm inclined to say this is okay for a minor version bump rather than major. If we think it's important to keep the old spelling to not break those using the client against ES then I can re-add the old property as deprecated in 1.x for removal in 2.x/main.

@`2.1`: https://github.com/opensearch-project/OpenSearch/blob/2.1/server/src/main/java/org/opensearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java#L73-L94
@`2.2`: https://github.com/opensearch-project/OpenSearch/blob/2.2/server/src/main/java/org/opensearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java#L73-L96

Reply:
Thank you for clarifying things!